configure.ps1

```diff
@@ -34,50 +31,139 @@ Param (
 
 . "$PSScriptRoot\build\common.ps1"
 
-Trace-Log "Configuring NuGet.Client build environment for VS${ToolsetVersion} Toolset"
+Trace-Log "Configuring NuGet.Client build environment"
+
+$BuildErrors = @()
 
-Update-SubModules -Force:$Force
+Invoke-BuildStep 'Configuring git repo' {
+    Update-SubModules -Force:$Force
+} -ev +BuildErrors
 
-Install-NuGet -Force:$Force
+Invoke-BuildStep 'Installing NuGet.exe' {
+    Install-NuGet -Force:$Force
+} -ev +BuildErrors
 
-Install-DotnetCLI -Force:$Force
+Invoke-BuildStep 'Installing .NET CLI' {
+    Install-DotnetCLI -Force:$Force
+} -ev +BuildErrors
 
-if ($CleanCache) {
+# Restoring tools required for build
+Invoke-BuildStep 'Restoring solution packages' {
+    Restore-SolutionPackages
+} -ev +BuildErrors
+
+Invoke-BuildStep 'Cleaning package cache' {
     Clear-PackageCache
+} -skip:(-not $CleanCache) -ev +BuildErrors
+
+$ConfigureObject = @{
+    BuildTools = @{}
+    Toolsets = @{}
 }
 
-Trace-Log "Validating VS${ToolsetVersion} toolset installation"
+Function New-BuildToolset {
+    param(
+        [int]$ToolsetVersion
+    )
+    $CommonToolsVar = "Env:VS${ToolsetVersion}0COMNTOOLS"
```

Comment:
This will be set in the Developer Command Prompt right? Is that a requirement? Or will this stuff also work from a vanilla PowerShell window?